src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -5861,11 +5862,16 @@ void MacroAssembler::zero_extend(Register dst, Register src, int bits) {
     default:
       break;
   }
-  slli(dst, src, XLEN - bits);
-  srli(dst, dst, XLEN - bits);
+
+  if (bits == 8) {
+    andi(dst, src, 0xFF);
```

Comment:
Maybe we could move `andi` to here:
https://github.com/openjdk/jdk/blob/6b022bb64b2109c8cd40ebd3b8b3226cf894544d/src/hotspot/cpu/riscv/macroAssembler_riscv.cpp#L5855-L5859

Reply:
Make sense. I just find that `zext_b(dst, src);` should not depend on `UseZbb`. So I just removed this check. Thanks.